### PR TITLE
EXPDAT-013 share product

### LINKED ISSUE #13: EXPDAT-013 share product
# share product

**As a** user

**I want** to share a product with my friends

**so that** he gets a message (whatsapp, email...) of a product I dont need with name and expiry date

# feature criteria
 - [x] every product in single view #8 has a share button
 - [x] by pressing the share button a windows should appear where I can choose how to send the message.
 - [x] when a messaging service is chosen a standard message like "Hi, I have {PRODUCT_NAME} that I don't need anymore and expires at {EXPIRY_DATE}. Do you want to have it?" should get inserted. The message is editable by the user.

# Acceptance criteria
- [ ] runs local without errors
- [ ] commit message contains issue abbreviations
- [ ] tests written for it

### PR DESCRIPTION
**Pull request by ChrisFartek**
This closes #13.

### CODE REVIEW

File: app/src/androidTest/java/at/sw21_tug/team_25/expirydates/DetailViewTest.kt [hunk 1]
```diff
@@ -1,8 +1,13 @@
 package at.sw21_tug.team_25.expirydates
 
 
+import android.app.Application
+import android.content.ClipboardManager
+import android.content.Context.CLIPBOARD_SERVICE
```
Comment (steschi) [inline app/src/androidTest/java/at/sw21_tug/team_25/expirydates/DetailViewTest.kt]: please remove unused imports

File: app/src/androidTest/java/at/sw21_tug/team_25/expirydates/DetailViewTest.kt [hunk 2]
```diff
@@ -161,6 +170,54 @@ class DetailViewTest {
 
     }
 
+    @Test
+    fun shareViewTest() {
+        val uiDevice = UiDevice.getInstance(InstrumentationRegistry.getInstrumentation())
+        val appContext = ApplicationProvider.getApplicationContext<Application>()
```
Comment (steschi) [inline app/src/androidTest/java/at/sw21_tug/team_25/expirydates/DetailViewTest.kt]: please remove unused variables

File: app/src/main/java/at/sw21_tug/team_25/expirydates/ui/detailview/ui/DetailViewActivity.kt [hunk 3]
```diff
@@ -56,13 +59,37 @@ class DetailViewActivity : AppCompatActivity() {
...
                     }
                 }
 
+                val shareButton = popupView.findViewById<Button>(R.id.share)
+                // set on-click listener
+                shareButton.setOnClickListener {
+                    var message = ""
+                    if (Util.getLanguage(activity as MainActivity) == Locale("en")) {
+                        message = "Hi, I have some $name that I don't need anymore and expires at $date. Do you want to have it?"
+                    } else if (Util.getLanguage(activity as MainActivity) == Locale("ru")) {
```
Comment (steschi) [inline app/src/main/java/at/sw21_tug/team_25/expirydates/ui/detailview/ui/DetailViewActivity.kt]: please move all hardcoded strings which need translation to the strings.xml

Details on how to store Format-Strings in resource files can be found here: https://developer.android.com/guide/topics/resources/string-resource#FormattingAndStyling

**Review by steschi — CHANGES_REQUESTED** [review]
The shareViewTest Testcase does not work for me. I am not sure if it is an issue on my emulator or not. 
image

**Review by florianmarcher — APPROVED** [review]
all tests passed

**Pull request merged**